quickwit/quickwit-indexing/src/actors/indexing_pipeline.rs

```diff
@@ -44,6 +45,11 @@ use crate::split_store::IndexingSplitStore;
 
 const MAX_RETRY_DELAY: Duration = Duration::from_secs(600); // 10 min.
 
+/// Spawning an indexing pipeline puts a lot of pressure on the file system, metastore, etc. so
```

Comment:
I'm not sure if putting a semaphore like this one is a good idea.

I expect to see a user starts an indexer and open an issue saying that it's not working.

Do you think we can put a limit right from the start of an indexer? So that we will tell the user that only the first 10 pipelines will run? One drawback of this approach is that if some pipelines can end, we end up by running less than 10 pipelines.

Reply:
This change does not limit the total number of indexing pipelines that can run on the node. This limits the number of indexing pipelines spawned concurrently on the node. The semaphore is only held while spawning the pipeline.

Reply:
Mmmm interesting, that's indeed a good idea, sorry for the confusion.

Reply:
Nitpick: the "See also #1638" is a bit frustrating as we want go click on it :)